### DISCUSSION TITLE
How can I connect a Node / Value from v1 to drive an animation in v2 ?

### DISCUSSION THREAD

**Discussion by andreibarabas** [software-mansion/react-native-reanimated#1559] [Q&A]
Hey guys,

**Context:**

We are using a bunch of libraries that rely on v1 to drive animation. One example being react-native-tab-view which we use directly for a custom tab experience.

The library exposes an animated node (position) that we currently use to animate styles of child components as they move in and out of the screen.

We would like to start migrating to v2, but don't know how to do it, as the position is a node on the v1 architecture and having to wait for every library to upgrade their dependency would probably be a high barrier of entry.

I welcome your thoughts on this matter. How would you solve it?

**Comment by jgonet** [comment]
v2 wasn't really designed to adapt to v1 but they can coexist in the same codebase.

You can try writing your own wrapper by utilizing [`__getValue()`](https://github.com/software-mansion/react-native-reanimated/blob/e9fe0c17bd1aef68f91fed4030898e95d1af907b/src/core/AnimatedNode.js#L121) function to get a current v1 value from the node and assign it to the shared value. This is unfortunately async so it may not help you much but it should give you some direction.

/cc @kmagiera, @karol-bisztyga, @Szymon20000

**Reply by andreibarabas** [reply]
I see @jakub-gonet . I’ll play with it a little.
